torch/csrc/jit/script/compiler.cpp

```diff
@@ -514,9 +514,9 @@ struct to_ir {
           ListAttributeMap list_attributes{};
           for (const auto& attr : apply.attributes()) {
             const auto& name = attr.name().name();
-            const Expr& value = attr.value();
+            const TreeRef& value = attr.value();
```

Comment:
Nooooo, why! This should never happen in a well formed AST

Reply:
@jamesr66a  - what was the bug prompting this error again? I imagine that there is an attribute node that is not an expression. We can either add it as an Expr and change the type back, or add an addition Attr type.

Reply:
Hey, having attributes as Expr's broke passing lists as attributes, which is super important. I believe test_view_shape_prop fails with the old statically-typed code

Reply:
We should add `List` to represent list literals in our AST (as a subclass of `Expr`) and use this instead